### PR TITLE
add yesterday section to time based switcher

### PR DESCRIPTION
**Pull request by chassu**
### How was this tested?
- [x] I tested this manually

## Issues addressed
#3823

### CODE REVIEW

File: Client/Frontend/Browser/Tabs/Tab.swift [hunk 1]
```diff
@@ -869,10 +873,16 @@ public func isLastExecutedTimeInTimeFilter(_ lastExecutedTime: Timestamp, _ byTi
             byAdding: .second, value: -15, to: Date())
         : Calendar.current.date(
             byAdding: .day, value: -1, to: Date())
-    let minusOneWeekToCurrentDate =
+    let minusTueDaysToCurrentDate =
```
Comment (evancrow) [inline Client/Frontend/Browser/Tabs/Tab.swift:876]: nit: Two (instead of Tue)?

File: Client/Frontend/Browser/Tabs/Tab.swift [hunk 2]
```diff
@@ -869,10 +873,16 @@ public func isLastExecutedTimeInTimeFilter(_ lastExecutedTime: Timestamp, _ byTi
...
         FeatureFlag[.shortenTimeThresholdForArchivingTabs]
         ? Calendar.current.date(
             byAdding: .second, value: -30, to: Date())
+        : Calendar.current.date(
```
Comment (evancrow) [inline Client/Frontend/Browser/Tabs/Tab.swift:880]: In PR #3824, which is still under review, I created a new helper function that you can use to simplify this create date code. Once that merges you will be able to refactor a lot of this down to `Date.getDate(dayOffset: -(days))`.

No action item here, just something for future cleanup

**Review by evancrow — APPROVED** [review]
LGTM, one nit

File: Client/Frontend/Browser/Tabs/Tab.swift (hunk 1 above)
Comment (chassu) [reply Client/Frontend/Browser/Tabs/Tab.swift]: can't believe I did that 😂

**Pull request merged**